config/config.go

```diff
@@ -320,6 +313,26 @@ func (c *Config) GetStatsFlushInterval() time.Duration {
 	return viper.GetDuration("stats.flush_interval")
 }
 
+// GetCommittedTierFSParams returns parameters for building a tierFS.  Caller must separately
+// build and populate Adaptor.
+func (c *Config) GetCommittedTierFSParams() (*pyramidparams.Params, error) {
+	adaptor, err := factory.BuildBlockAdapter(c)
```

Comment:
This is changing the current pattern of block adapter being a singleton, is it desired?
I can imagine a scenario where we want just a single instance to control throughputs, retries, etc ..

You can still use the factory and have a singleton if you save the first instance you created and return it for secondary requests.

Reply:
Not done.  Singletons may or may not be needed here, the fact we currently have just the one block adapter seems just a historical trivium to me, with no rationale one way or the other.  I can imagine all of these use-cases, and cannot prefer one to another right now:
1. Everything entirely separate.
1. Some adapters separate, some shared, some separate with shared component.  (E.g. S3 would be a good candidate for this!)
1. One adapter of each type.
1. One adapter for everything.

I really hate infrastructure that forces singletons on me.  E.g. @johnnyaug had some problems last and this week because there can only be one S3a endpoint in a particular software version.

So until I know better, I want to go with the least effort -- which is what I implements (because I was lazy and didn't even think about it).